opentelemetry-http/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+/// allows users to bring their choice of http clients.
+#[async_trait]
+pub trait HttpClient: Debug + Send + Sync {
+    /// Send a batch of spans to collectors
+    async fn send(&self, request: Request<Vec<u8>>) -> Result<(), TraceError>;
+}
+
+#[cfg(feature = "reqwest")]
+mod reqwest {
+    use super::{async_trait, HttpClient, Request, TraceError};
+    use opentelemetry::sdk::export::ExportError;
+    use std::convert::TryInto;
+    use thiserror::Error;
+
+    #[async_trait]
+    impl HttpClient for reqwest::Client {
+        async fn send(&self, request: Request<Vec<u8>>) -> Result<(), TraceError> {
+            let request = request.try_into().map_err(ReqwestError::from)?;
+            let _ = self
+                .execute(request)
+                .await
+                .and_then(|rsp| rsp.error_for_status())
+                .map_err(ReqwestError::from)?;
+            Ok(())
+        }
+    }
+
+    #[async_trait]
+    impl HttpClient for reqwest::blocking::Client {
+        async fn send(&self, request: Request<Vec<u8>>) -> Result<(), TraceError> {
+            let _ = request
+                .try_into()
+                .and_then(|req| self.execute(req))
+                .and_then(|rsp| rsp.error_for_status())
+                .map_err(ReqwestError::from)?;
+            Ok(())
+        }
+    }
+
+    #[derive(Debug, Error)]
+    #[error(transparent)]
+    struct ReqwestError(#[from] reqwest::Error);
+
+    impl ExportError for ReqwestError {
+        fn exporter_name(&self) -> &'static str {
+            "reqwest"
+        }
+    }
+}
+
+#[cfg(feature = "surf")]
+mod surf {
+    use super::{async_trait, HttpClient, Request, TraceError};
+    use opentelemetry::sdk::export::ExportError;
+    use std::fmt::{Display, Formatter};
+
+    #[async_trait]
+    impl HttpClient for surf::Client {
+        async fn send(&self, request: Request<Vec<u8>>) -> Result<(), TraceError> {
+            let (parts, body) = request.into_parts();
+            let uri = parts
+                .uri
+                .to_string()
+                .parse()
+                .map_err(|_err: surf::http::url::ParseError| TraceError::from("error parse url"))?;
+
+            let req = surf::Request::builder(surf::http::Method::Post, uri)
+                .content_type("application/json")
+                .body(body);
+            let result = self.send(req).await.map_err::<SurfError, _>(Into::into)?;
+
+            if result.status().is_success() {
+                Ok(())
+            } else {
+                Err(SurfError(surf::Error::from_str(
+                    result.status(),
+                    result.status().canonical_reason(),
+                ))
+                .into())
+            }
+        }
+    }
+
+    #[derive(Debug)]
+    struct SurfError(surf::Error);
+
+    impl ExportError for SurfError {
+        fn exporter_name(&self) -> &'static str {
+            "surf"
+        }
+    }
+
+    #[cfg(feature = "surf")]
```

Comment:
Do we still need this even the mod is guarded by a `#[cfg(feature = "surf")]`?

Reply:
We don't, good catch! Fixed it in the latest version.